src/QsCompiler/BondSchemas/EntryPoint/EntryPoint.bond

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.QsCompiler.BondSchemas.EntryPoint;
+
+enum DataType
+{
+    BoolType,
+    IntegerType,
+    DoubleType,
+    PauliType,
+    RangeType,
+    ResultType,
+    StringType,
+    ArrayType
+}
+
+enum PauliValue
+{
+    PauliI,
+    PauliX,
+    PauliY,
+    PauliZ
+}
+
+struct RangeValue
+{
+    5: int64 Start;
+
+    10: int64 Step;
+
+    15: int64 End;
+}
+
+enum ResultValue
+{
+    Zero,
+    One
+}
+
+struct ArrayValue
+{
+    5: DataType Type = BoolType;
+
+    // Not null when Type is BoolType.
+    10: nullable<vector<bool>> Bool;
+
+    // Not null when Type is IntegerType.
+    15: nullable<vector<int64>> Integer;
+
+    // Not null when Type is DoubleType.
+    20: nullable<vector<double>> Double;
+
+    // Not null when Type is PauliType.
+    25: nullable<vector<PauliValue>> Pauli;
+
+    // Not null when Type is RangeType.
+    30: nullable<vector<RangeValue>> Range;
+
+    // Not null when Type is ResultType.
+    35: nullable<vector<ResultValue>> Result;
+}
+
+struct ArgumentValue
+{
+    5: DataType Type = BoolType;
+
+    // Not null when Type is BoolType.
+    10: nullable<bool> Bool;
+
+    // Not null when Type is IntegerType.
+    15: nullable<int64> Integer;
+
+    // Not null when Type is DoubleType.
+    20: nullable<double> Double;
+
+    // Not null when Type is PauliType.
+    25: nullable<PauliValue> Pauli;
+
+    // Not null when Type is RangeType.
+    30: nullable<RangeValue> Range;
+
+    // Not null when Type is ResultType.
+    35: nullable<ResultValue> Result;
+
+    // Not null when Type is StringType.
+    40: nullable<string> String;
+
+    // Not null when Type is ArrayType.
+    45: nullable<ArrayValue> Array;
+}
+
+struct Argument
+{
+    5: DataType Type = BoolType;
```

Comment:
`Argument` has a type but so does `ArgumentValue`. Since tuples are not supported on entrypoints and arrays must always be of the same type, it seems like only the type and array type at `Argument` should be necessary.

Reply:
I was thinking that having each structure standalone but in this case it does not really make sense to have a `Type` member in `ArgumentValue`. I'll remove it.

Reply:
Removed the redundant members.